src/Sulu/Bundle/ContentBundle/Tests/Unit/Repository/ResourceLocatorRepositoryTest.php

```diff
@@ -0,0 +1,243 @@
+
+use Sulu\Bundle\ContentBundle\Document\BasePageDocument;
+use Sulu\Bundle\ContentBundle\Document\PageDocument;
+use Sulu\Bundle\ContentBundle\Repository\ResourceLocatorRepository;
+use Sulu\Bundle\ContentBundle\Repository\ResourceLocatorRepositoryInterface;
+use Sulu\Component\Content\Compat\Property;
+use Sulu\Component\Content\Compat\StructureInterface;
+use Sulu\Component\Content\Compat\StructureManagerInterface;
+use Sulu\Component\Content\Types\Rlp\ResourceLocatorInformation;
+use Sulu\Component\Content\Types\Rlp\Strategy\RlpStrategyInterface;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+
+/**
+ * @group unit
+ * @group repository
+ */
+class ResourceLocatorRepositoryTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ResourceLocatorRepositoryInterface
+     */
+    private $repository;
+
+    /**
+     * @var StructureManagerInterface
+     */
+    private $structureManager;
+
+    /**
+     * @var RlpStrategyInterface
+     */
+    private $rlpStrategy;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    protected function setUp()
+    {
+        $this->structureManager = $this->prophesize(StructureManagerInterface::class);
+        $this->rlpStrategy = $this->prophesize(RlpStrategyInterface::class);
+        $this->documentManager = $this->prophesize(DocumentManagerInterface::class);
+
+        $this->repository = new ResourceLocatorRepository(
+            $this->rlpStrategy->reveal(),
+            $this->structureManager->reveal(),
+            $this->documentManager->reveal()
+        );
+    }
+
+    public function testGenerateWithParentUuid()
+    {
+        // method parameter
+        $parts = [
+            'title' => 'news',
+            'subtitle' => 'football',
+        ];
+        $parentUuid = '0123456789abcdef';
+        $webspace = 'sulu_io';
+        $locale = 'en';
+        $template = 'default';
+
+        // template structure
+        $structure = $this->prophesize(StructureInterface::class);
+        $structure->getPropertiesByTagName('sulu.rlp.part')->willReturn([
+            new Property('subtitle', 'subtitle', 'subtitle'),
+            new Property('title', 'title', 'title'),
+        ]);
+        $this->structureManager->getStructure($template)->willReturn($structure->reveal());
+
+        // parent document
+        $parent = $this->prophesize(BasePageDocument::class);
+        $parent->getPublished()->willReturn(true);
+        $this->documentManager->find($parentUuid, $locale, ['load_ghost_content' => false])->willReturn($parent->reveal());
+
+        // path related
+        $parentPath = '/parent';
+        $this->rlpStrategy->loadByContent($parent)->willReturn($parentPath);
+
+        $title = 'news-football';
+        $path = $parentPath . '/' . $title;
+        $this->rlpStrategy->generate($title, $parentPath, $webspace, $locale, null)->willReturn($path);
+
+        // method call
```

Comment:
Actually almost all comments in this file could be removed... Please only add comments if they add real value. If you create a document and assign it to the variable `$parent` it's obvious that it is a parent document. If you add a comment saying that it's just duplicating information, which might even diverge somewhen in the future.